### PR TITLE
Fix a bug that was silencing errors from the server.

### PR DESCRIPTION
**Pull request by zenhack**
I caught this while working on OBMd stuff, which raises another question: right now OBMd doesn't follow the same error reporting convention as the HIL api server.

### CODE REVIEW

**Comment by naved001** [comment]
If it's simple enough, can we write a unit test for this?

**Comment by zenhack** [comment]
Hm, I think that's worth doing, but thinking about how it's a little fiddly. I'd like to wait until #1009 is merged, since I'll need to use the test client, and don't really want to deal with potential conflicts. I'll probably end up moving that into test_common, and have the test for this in a separate file from the existing client (integration) tests. I think I'll tag this one as "waiting on other change."

**Comment by zenhack** [comment]
Added a test.

**Comment by zenhack** [comment]
(Per the docstring at the top of the file, we'll want to merge your thing and then rename it).

File: tests/unit/client_regression.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from schema import Schema
+
+from hil import config, rest
+from hil.client.base import FailedAPICallException
+from hil.client.client import Client
+from hil.test_common import HybridHTTPClient, fail_on_log_warnings, \
+    fresh_database, server_init, config_testsuite
+
+fail_on_log_warnings = pytest.fixture(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
```
Comment (naved001) [inline tests/unit/client_regression.py]: why is this fixture required?

Comment (zenhack) [reply tests/unit/client_regression.py]: It isn't, I just add the "standard" fixtures without really thinking about it -- I've removed it.

**Comment by Izhmash** [comment]
@zenhack we'll need to fix the merge conflicts here

**Comment by zenhack** [comment]
Rebased. I also did the rename suggested in the docstring, now @naved001's pr is merged.

**Comment by naved001** [comment]
Cool, merging.

**Pull request merged**